elastic/kibana
# [TableListView] Fix tag selection when passing initialFilter

In https://github.com/elastic/kibana/pull/160540 we discovered that the tag selection was not applied correctly when passing an `initialFilter` to the TableListView.

This PR fixes this behaviour.

While debugging I also found another bug in the DashboardListing page, the "clear tag selection" was not working correctly because of an unstable reference to the saved object tagging plugin.

## How to test

* Install one of the sample data set
* Go to dashboard listing, and edit one of them --> add a tag to the dashboard (e.g. "foo")
* Go back the listing.
* Modify the `dashboard_listing.tsx` file (src/plugins/dashboard/public/dashboard_listing/dashboard_listing.tsx)

```diff
--- a/src/plugins/dashboard/public/dashboard_listing/dashboard_listing.tsx
+++ b/src/plugins/dashboard/public/dashboard_listing/dashboard_listing.tsx
@@ -240,7 +240,8 @@ export const DashboardListing = ({
           title={getTableListTitle()}
           headingId="dashboardListingHeading"
           initialPageSize={initialPageSize}
-          initialFilter={initialFilter}
+          // initialFilter={initialFilter}
+          initialFilter={`tag:("foo")`}
           entityName={getEntityName()}
           listingLimit={listingLimit}
           emptyPrompt={emptyPrompt}
```
* Refresh the page
* The search text should be updated and the tag selection should be correct

<img width="1282" alt="Screenshot 2023-06-29 at 12 35 50" src="https://github.com/elastic/kibana/assets/2854616/d5581a27-3fa6-4ea2-9924-9c7934290843">
